### PR TITLE
feat: set expected entity when capturing inputs to API calls in .dialog import

### PR DESCRIPTION
**Pull request by tpmsr**
For cases where a bot asks the user a question (eg "_What is your product key?_") whose response will be used as the input to an API call, this PR updates the import flow to 

1. Create an entity to capture the results of the user response (eg, _`productKey`_)
2. Sets expected entity on the action preceding the user response to the name of the entity created in step 1

We later look up the entity by its name when creating actions, and set the expected / suggested entity on the appropriate action.

Tested with the license key bot :

expected_entity

### CODE REVIEW

File: src/routes/Apps/App/TrainDialogs.tsx [hunk 1]
```diff
@@ -157,6 +157,8 @@ interface TranscriptImportData {
     warnings: string[]
     // Conditions are keyed by TrainScorerStep.importId.
     conditions?: { [key: string]: CLM.Condition[] }
+    // Expected entities are keyed by actionId.
+    expectedEntityNames?: { [key: string]: string }
```
Comment (mattmazzola) [inline src/routes/Apps/App/TrainDialogs.tsx]: I would prefer names like `actionIdToEntityNameMap`.  Then you don't need the comment as it's self explanatory.  As it is the name is misleading.

I used names that are plural to mean it's a list. 
```
entityName: 'myEntity',
entityNames: ['entityOne', 'entityTwo']
```

This name makes it sound like it should be a list of strings, but it's not.

File: src/Utils/obiDialogParser.ts [hunk 2]
```diff
@@ -504,7 +553,7 @@ export class ObiDialogParser {
         // any collisions.
         let updatedValues: { [key: string]: Set<string> } = {}
         for (const value of values) {
-            const normalized = this.normalizeEnumValueName(value)
+            const normalized = this.normalizeValueName(value, CLM.MAX_ENUM_VALUE_NAME_LENGTH)
```
[the remark this answers is not in the document]
Comment (tpmsr) [reply src/Utils/obiDialogParser.ts]: Not sure I understand your comment - this is called on line 522 for entities, with a constant reflecting entity name length, and on 556 for enums, with a constant reflecting the enum length.  Is there some issue with this ?

File: src/Utils/obiDialogParser.ts [hunk 3]
```diff
@@ -545,6 +594,34 @@ export class ObiDialogParser {
         return newEntity
     }
 
+    private async createCustomTrainedEntity(entityName: string): Promise<CLM.EntityBase> {
+        const newEntity: CLM.EntityBase = {
+            entityId: undefined!,
+            entityName,
+            resolverType: "none",
```
[the remark this answers is not in the document]
Comment (tpmsr) [reply src/Utils/obiDialogParser.ts]: Updated

File: src/routes/Apps/App/TrainDialogs.tsx (hunk 1 above)
Comment (tpmsr) [reply src/routes/Apps/App/TrainDialogs.tsx]: Done

**Pull request merged**